CHANGELOG.md

```diff
@@ -180,6 +180,13 @@ Adding a new version? You'll need three changes:
 - Admin and proxy listens in the deploy manifests now use the same parameters
   as the default upstream kong.conf.
   [#3165](https://github.com/Kong/kubernetes-ingress-controller/pull/3165)
+- Fix the behavior of filtering hostnames in `HTTPRoute` when listeners 
+  of parent gateways specified hostname. Now if an `HTTPRoute` does not 
+  specify hostnames, and one of its parent listeners has not specified
+  hostname, the hostnames will not be filtered in `HTTPRoute`. 
```

Comment:
Is there any chance we use a different wording for this? It's feel way to complicated for me and I'm having a hard time understanding what it really means.

As for the use of word filtered: I'm not a huge fan because it's awfully ambiguous (at least for me). Are we filtering in (retaining) or filtering out (removing) the hostnames?

Reply:
Updated. Hope the new version could be easier to understand.